libavcodec/hevc_mp4toannexb_bsf.c

```diff
@@ -155,10 +155,14 @@ static int hevc_mp4toannexb_filter(AVBSFContext *ctx, AVPacket *out)
         }
 
         nalu_type = (bytestream2_peek_byte(&gb) >> 1) & 0x3f;
+        has_vps = (has_vps || nalu_type == HEVC_NAL_VPS);
+        has_sps = (has_sps || nalu_type == HEVC_NAL_SPS);
+        has_pps = (has_pps || nalu_type == HEVC_NAL_PPS);
 
         /* prepend extradata to IRAP frames */
         is_irap       = nalu_type >= 16 && nalu_type <= 23;
-        add_extradata = is_irap && !got_irap;
+        /* ignore the extradata if IRAP frame has vps, sps and pps */
+        add_extradata = is_irap && !got_irap && (!has_vps || !has_sps || !has_pps);
```

Comment:
the comment may not match the code. The code means, once we see sps, pps in the stream, we will not add extra data.

Reply:
The extradata is ignored if a IRAP frame already has all of vps, sps and pps, so we won't prepend the data to out AVPacket. I think it is the same of what you said.